modules/Functions/Functions.Rmd

```diff
@@ -443,12 +443,42 @@ airquality %>%
 
 Similar to across, `purrr` is a package that allows you to apply a function to multiple columns in a data frame or multiple data objects in a list.
 
+## map
 
-## map_df
+By default, the map functions return lists. So in this example we have to follow up `map()` with `bind_cols()` to turn our `list` back into a `data.frame`.  
 
 ```{r}
 library(purrr)
-airquality %>% map_df(replace_na, replace = 0)
+airquality %>% 
+  map(replace_na, replace = 0) %>% 
+  bind_cols()
+```
+Note you could accomplish the same task by doing: `airquality %>% mutate_all(~replace_na(., replace = 0))`
+
+## pmap
+
+`pmap()` is a very handy `purrr` function for running through multiple inputs at once. Let's say we wanted to do something different with each variable. 
+
+```{r}
```

Comment:
This example is almost certainly too overwhelming, but I do think pmap is cool, so consider if you want to add it (`pmap` is great for people who are scared of `for` loops).

Reply:
I agree pmap is cool! I wonder if there's a different example we could use. I think students will be confused why they should use pmap in this case rather than
`airquality %>% mutate(wind_metric = Wind * 1.609)` for example. 

This example from the docs is pretty nice
```
df <- data.frame(
  x = c("apple", "banana", "cherry"),
  pattern = c("p", "n", "h"),
  replacement = c("P", "N", "H"),
  stringsAsFactors = FALSE
  )
pmap(df, gsub)
```
Maybe we could play with something that replaces spaces, eg. `str_replace_all(column, " ", "_")` on all columns of `cars` without having to repeat `mutate` statements?

Reply:
Actually across would probably be most efficient here.
```
cars <- cars %>%
  mutate(across(c(Model, SubModel, Nationality, Size), ~str_replace_all(., " ", "_")))
```
Let's take out `pmap` unless we can come up with an example where it saves line space or does something very different.